src/ripple/consensus/Validations.h

```diff
@@ -590,9 +606,45 @@ class Validations
             // validations sequence from that validator
             auto const now = byLedger_.clock().now();
             SeqEnforcer<Seq>& enforcer = seqEnforcers_[nodeID];
+
             if (!enforcer(now, val.seq(), parms_))
                 return ValStatus::badSeq;
 
+            {
+                auto& vs = bySequence_[val.seq()];
+
+                auto const [it, inserted] = vs.emplace(nodeID, val);
+
+                if (!inserted)
+                {
+                    // Check if it's expired; if it is, replace it. We need to
+                    // calculate the difference between the sign times for the
+                    // two validations:
+                    auto const diff =
+                        std::max(it->second.signTime(), val.signTime()) -
+                        std::min(it->second.signTime(), val.signTime());
+
+                    if (diff > std::chrono::minutes{10} &&
+                        val.signTime() > it->second.signTime())
+                        it->second = val;
+                }
+
+                // If the ledger hash matches, it must be a validation for
+                // the same ledger:
+                bool const same = (it->second.ledgerID() == val.ledgerID());
```

Comment:
I'm unclear how these conditions wouldn't already be blocked by the `badSeq` one above. That is, the `enforcer` requires a strictly increasing sequence number seen by a validator. So I think it already filters these alternate outcomes.

Reply:
If you do want to check these, an option would be to move them above before `badSeq` returns.

Reply:
Interesting comment here @bachase . I would argue that this is a valid check due to malicious/byzantine behavior. An attacker can increment the seq without changing the content. So this check can help us catch those byzantine behaviors is what I am thinking. I was thinking maybe the signature check would block the logic here but it seems to me that sfLedgerHash is just one field within the digest so a byzantine node can still sign the same validation with a different sequence and this logic here would catch it

Reply:
@carlhua but doesn't this check presume the two validations being compared have the same sequence number? They both come from the `bySequence_` map in [Line 614](https://github.com/ripple/rippled/pull/3291/files/31796963fa7ca6226751366b78d324775f0fe584..330f35b5f5712c758057fed0a962d5d831960ad8#diff-46f83c3cd15dd825b63068c0ab4dea00R616).

Reply:
@bachase you are right!!  These checks would still guard against the scenarios depicted in the code right?
1. Same sequence, hashs match but cookies don't so we say its multiple
2. Same sequence, hashes don't match, we say its conflicting

Reply:
I sorted this out already. I this I misfolded some of my work before pushing. See updated commit.